### PR TITLE
Add claim operation to the `fungible` app

### PR DESCRIPTION
**Pull request by ma2bd**
This is similar to the system operation.

One minor difference is that `AccountOwner`s can be applications here but this use case is not yet supported. (Authenticated caller is not forwarded through effects yet: #708)

### CODE REVIEW

File: linera-core/src/unit_tests/client_tests.rs [hunk 1]
```diff
@@ -308,7 +308,7 @@ where
     assert!(sender.key_pair().await.is_ok());
     assert_eq!(
         builder
-            .check_that_validators_have_certificate(sender.chain_id, BlockHeight::from(0), 4)
+            .check_that_validators_have_certificate(sender.chain_id, BlockHeight::from(0), 3)
```
Comment (MathieuDutSik) [inline linera-core/src/unit_tests/client_tests.rs:311]: Why this change

Comment (ma2bd) [reply linera-core/src/unit_tests/client_tests.rs:311]: It's a quick CI fix [fix flaky assert in test_rocksdb_share_ownership](https://github.com/linera-io/linera-protocol/pull/703/commits/4d2460c4f7e3ff498796dbe31bff4d32534efdf4)

File: linera-service/tests/end_to_end_tests.rs [hunk 2]
```diff
@@ -1282,20 +1282,22 @@ async fn test_end_to_end_fungible() {
...
+    let source = format!(
         "{{chainId: \"{}\", owner: {}}}",
         chain1,
         account_owner2.to_value()
     );
+    let destination = format!(
+        "{{chainId: \"{}\", owner: {}}}",
+        chain2,
+        account_owner2.to_value()
+    );
```
Comment (MathieuDutSik) [inline linera-service/tests/end_to_end_tests.rs:1295]: That pattern of building the address repeats quite a lot and maybe deserve its own function to avoid repetition.

File: linera-service/tests/end_to_end_tests.rs [hunk 3]
```diff
@@ -1308,7 +1310,7 @@ async fn test_end_to_end_fungible() {
     let value = app1
         .get_fungible_account_owner_amount(&account_owner2)
         .await;
-    assert_eq!(value, Amount::from(3));
```
Comment (MathieuDutSik) [inline linera-service/tests/end_to_end_tests.rs:1311]: for the control check, maybe best would be a function taking an application and a btreemap. that would be nicer and easier to read.

Comment (ma2bd) [reply linera-service/tests/end_to_end_tests.rs:1311]: In tests, the code is usually more streamlined (less intermediate functions) precisely to make it easier to remove

File: linera-service/tests/end_to_end_tests.rs (hunk 2 above)
Comment (ma2bd) [reply linera-service/tests/end_to_end_tests.rs:1295]: Sure, this one I can do

Comment (ma2bd) [reply linera-service/tests/end_to_end_tests.rs:1295]: It was your code though :)

Comment (ma2bd) [reply linera-service/tests/end_to_end_tests.rs]: Done. The best solution is actually to use `fungible::Account`.

File: linera-service/tests/end_to_end_tests.rs (hunk 3 above)
Comment (ma2bd) [reply linera-service/tests/end_to_end_tests.rs:1311]: Just to be clear: for me, the current code is fine here. Another way to write this assertion (also fine) is:
```
assert_eq!(
    app1.get_fungible_account_owner_amount(&account_owner2).await,
    Amount::from(3),
);
```
